homeassistant/components/switch/rachio.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import voluptuous as vol
+from datetime import timedelta
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+REQUIREMENTS = ['https://github.com/Klikini/rachiopy'
+                '/archive/edf666de8ef3f9596ddc8d3a989e8c4b829a4319.zip'
+                '#rachiopy==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'switch'
+
+DATA_RACHIO = 'rachio'
+
+CONF_manual_run_mins = 'manual_run_mins'
+manual_run_mins = 60
+
+STATUS_ONLINE = 'ONLINE'
+
+MIN_UPDATE_INTERVAL = timedelta(minutes=5)
+MIN_FORCED_UPDATE_INTERVAL = timedelta(seconds=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string
+})
+
+_IRO = None
+
+
+# Set up the component
+# noinspection PyUnusedLocal
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    global _IRO
+    global manual_run_mins
+
+    # Get options
+    manual_run_mins = config.get(CONF_manual_run_mins) or manual_run_mins
+    _LOGGER.debug("Rachio run time is " + str(manual_run_mins) + " min")
+
+    # Get access token
+    _LOGGER.debug("Getting Rachio access token...")
+    access_token = config.get(CONF_ACCESS_TOKEN)
+    if not access_token:
+        _LOGGER.error("Rachio API access token must be set "
+                      "in the configuration")
+        return False
+
+    # Configure API
+    _LOGGER.debug("Configuring Rachio API...")
+    from rachiopy import Rachio
+    r = Rachio(access_token)
+    person = _get_person(r)
+
+    # Get and persist devices
+    devices = _list_devices(r)
+    if len(devices) == 0:
+        _LOGGER.error("No Rachio devices found in account " +
+                      person['username'])
+        return False
+    else:
+        _IRO = devices[0]
+
+        if len(devices) > 1:
+            _LOGGER.warning("Multiple Rachio devices found in account, "
+                            "using " + _IRO.device_id)
+        else:
+            _LOGGER.info("Found Rachio device")
+
+    _IRO.update()
+    add_devices(_IRO.list_zones())
+    return True
+
+
+# Pull the account info of the person whose access token was provided
+def _get_person(r):
+    person_id = r.person.getInfo()[1]['id']
+    return r.person.get(person_id)[1]
+
+
+# Pull a list of devices on the account
+def _list_devices(r):
+    return [RachioIro(r, d['id']) for d in _get_person(r)['devices']]
+
+
+# Represents one Rachio Iro
+class RachioIro(object):
```

Comment:
RachioPy is not my own creation, and in fact includes these classes, however, they only contain static methods and are otherwise useless. I'll add these methods to my fork of the library.